source/core/index.ts

```diff
@@ -793,6 +793,10 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			return;
 		}
 
+		// `HTTPError`s always have `error.response.body` defined.
+		// Therefore we cannot retry if `options.throwHttpErrors` is false.
+		// On the last retry, if `options.throwHttpErrors` is false, we would need to return the body,
+		// but that wouldn't be possible since the body would be already read in `error.response.body`.
```

Comment:
If I'm understanding this correctly, it cannot be read multiple times because it's a stream, right? Maybe we could buffer the stream and fake it somehow?

If not, we need to at least document the behavior.

Reply:
> If I'm understanding this correctly, it cannot be read multiple times because it's a stream, right?

Correct.

> Maybe we could buffer the stream and fake it somehow?

That's extremely hacky and would result in double the RAM usage. I'm not saying I'm against it, I'd just prefer if we didn't do that. For example, if we get 1GB of data, RAM usage would be 2GB since we would have a string form and a buffer form.

Although we already double RAM usage for Promise API when `responseType` is set to something different to `string`.

Reply:
It could be stored as buffer though and then if string was needed, it could be a getter. But I agree it may not be worth the complication, so I suggest we just document the behavior.

Reply:
If it's a getter then it won't be visible in logs :/

Reply:
Getters are not enumerable by default, but you can make a getter enumerable.

Reply:
Wouldn't it look like this:

```
{
body: [ getter: body ]
}
```

?

Reply:
https://stackoverflow.com/questions/28072671/how-can-i-get-console-log-to-output-the-getter-result-instead-of-the-string-ge

Reply:
Ah yes, forgot about `util.inspect.custom` - however that would be a hack on top of another hack :joy: not worth it imo

Reply:
I agree. I think it's enough to just do the mentioned docs changes.